test/testclass.cpp

```diff
@@ -908,6 +925,14 @@ class TestClass : public TestFixture {
         ASSERT_EQUALS("", errout.str());
     }
 
+    void noConstructorButEqtoSelf() {
+        checkConstructor("struct foobar {\n"
+            "   auto operator=(const foobar&)->foobar&;\n"
+            " };\n"
+            "auto foobar::operator=(const foobar&)->foobar & = default;\n");
```

Comment:
I think the real point here is to ensure that the `Tokenizer/SymbolDatabase` will handle this method properly. Where is the syntax error thrown? If it's thrown in `SymbolDatabase` then I think a test case in `TestSymbolDatabase` would make sense.

Reply:
like written above it is this throw:
```cpp
void CheckClass::initializeVarList(const Function &func, std::list<const Function *> &callstack, const Scope *scope, std::vector<Usage> &usage)
{
    if (!func.functionScope)
        throw InternalError(nullptr, "Internal Error: Invalid syntax"); // #5702
```

Check my commend below (in a second) I found something...

Reply:
hmm.. I am not sure what the proper data is off the top of my head. It sounds reasonable that functionScope is null.

It seems strange that we throw such exception in CheckClass.. if validation is needed it would be better to do it in SymbolDatabase.

Reply:
The test case that was added for this exception looks like garbage to me: 623e5db0b2601e1cd08e44dfdb90cb1cfffb1bd3

Somebody should try to implement that more properly. That test case belongs in testgarbage as far as I see and the syntax error should be thrown in tokenizer/symboldatabase.

Reply:
Well it turned out to be an issue in the symboldatabase...